packages/react-hooks/src/useSearch/useSearch.ts

```diff
@@ -44,34 +52,41 @@ export function useSearchOne<K extends ResourceType>(
  *
  * @param resourceType - The FHIR resource type to search.
  * @param query - Optional search parameters.
+ * @param options - Optional options for configuring the search.
  * @returns A 3-element tuple containing the search result, loading flag, and operation outcome.
  */
 export function useSearchResources<K extends ResourceType>(
   resourceType: K,
-  query?: QueryTypes
+  query?: QueryTypes,
+  options?: SearchOptions
 ): [ResourceArray<ExtractResource<K>> | undefined, boolean, OperationOutcome | undefined] {
-  return useSearchImpl<K, ResourceArray<ExtractResource<K>>>('searchResources', resourceType, query);
+  return useSearchImpl<K, ResourceArray<ExtractResource<K>>>('searchResources', resourceType, query, options);
 }
 
-function useSearchImpl<K extends ResourceType, ReturnType>(
+function useSearchImpl<K extends ResourceType, SearchReturnType>(
   searchFn: SearchFn,
   resourceType: K,
-  query: QueryTypes | undefined
-): [ReturnType | undefined, boolean, OperationOutcome | undefined] {
+  query: QueryTypes | undefined,
+  options?: SearchOptions
+): [SearchReturnType | undefined, boolean, OperationOutcome | undefined] {
   const medplum = useMedplum();
-  const [searchKey, setSearchKey] = useState<string>();
+  const [lastSearchKey, setLastSearchKey] = useState<string>();
   const [loading, setLoading] = useState<boolean>(true);
-  const [result, setResult] = useState<ReturnType>();
+  const [result, setResult] = useState<SearchReturnType>();
   const [outcome, setOutcome] = useState<OperationOutcome>();
 
+  const searchKey = medplum.fhirSearchUrl(resourceType, query).toString();
+  const [debouncedSearchKey] = useDebouncedValue(searchKey, options?.debounceMs ?? DEFAULT_DEBOUNCE_MS, {
```

Comment:
Adding debouncing by default feels like too big of a breaking change. I think we should have it disabled by default and make a v4 issue to enable debouncing by default.

Reply:
I don't totally agree that this is breaking: I would argue that instant search queries for all inputs isn't really a part of the contract of the hook

Reply:
As discussed offline, concerns about breaking customer functionality are fair. We decided to drop the default debounce time to 250ms, since it's much less likely to cause perceivable differences in functionality while still retaining some inherent performance wins for the default settings.

Also worth pointing out that we are using the `leading` option for the `useDebouncedValue` hook from `@mantine/hooks` which applies the first change in the value in a string of updates immediately, so that means for only one change in the search, the search is still immediate and it's only when the query parameters change rapidly that they are debounced by the 250ms.